src/main/java/de/blackforestsolutions/datamodel/Leg.java

```diff
@@ -0,0 +1,264 @@
+        for (Field attributeToCheck : this.getClass().getDeclaredFields()) {
+            try {
+                if (attributeToCheck.get(this) != null) {
+                    hashCodeBuilder.append(attributeToCheck.hashCode());
+                }
+            } catch (IllegalAccessException e) {
+                log.error("Access Error while accessing to Travelpoint", e);
+            }
+        }
+        return hashCodeBuilder.toHashCode();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Leg leg = (Leg) o;
+        return Objects.equals(id, leg.id)
+                &&
+                Objects.equals(start, leg.start)
+                &&
+                Objects.equals(destination, leg.destination)
+                &&
+                Objects.equals(startTime, leg.getStartTime())
+                &&
+                Objects.equals(arrivalTime, leg.arrivalTime)
+                &&
+                Objects.equals(duration, leg.duration)
+                &&
+                Objects.equals(delay, leg.delay)
+                &&
+                Objects.equals(travelLine, leg.travelLine)
+                &&
+                Objects.equals(price, leg.price)
+                &&
+                Objects.equals(distance, leg.distance)
+                &&
+                Objects.equals(providerId, leg.providerId)
+                &&
+                Objects.equals(travelProvider, leg.travelProvider)
+                &&
+                Objects.equals(unknownTravelProvider, leg.unknownTravelProvider)
+                &&
+                Objects.equals(vehicleType, leg.vehicleType)
+                &&
+                Objects.equals(vehicleName, leg.vehicleName)
+                &&
+                Objects.equals(vehicleNumber, leg.vehicleNumber)
+                &&
+                Objects.equals(incidents, leg.incidents);
+
+    }
+
+
+    @Setter
+    @Getter
+    @JsonPOJOBuilder(withPrefix = "set")
+    @NoArgsConstructor(access = AccessLevel.PRIVATE)
+    public static class LegBuilder {
+
+        @Id
+        private UUID id;
+
+        private TravelPoint start;
+
+        private TravelPoint destination;
+
+        private Date startTime;
+
+        private Date arrivalTime;
+
+        private Duration duration;
+
+        private Duration delay;
+
+        private TravelLine travelLine;
+
+        private Price price;
+
+        private Distance distance;
+
+        private String providerId = "";
+
+        private TravelProvider travelProvider;
+
+        private String unknownTravelProvider = "";
+
+        private VehicleType vehicleType;
+
+        private String vehicleName = "";
+
+        private String vehicleNumber = "";
+
+        private List<String> incidents = new ArrayList<>();
+
+        public LegBuilder(UUID id) {
```

Comment:
We need a kind of copy constructor here dont we (leg to leg builder ) for the case that we would like to change something.